src/ert/_c_wrappers/enkf/ensemble_config.py

```diff
@@ -246,21 +247,29 @@ def gen_data_node(gen_data: List[str]) -> Optional[GenDataConfig]:
         report_steps = rangestring_to_list(options.get(ConfigKeys.REPORT_STEPS, ""))
 
         if os.path.isabs(res_file) or "%d" not in res_file:
-            logger.error(
-                f"The RESULT_FILE:{res_file} setting for {name} is invalid - "
-                "must have an embedded %d - and be a relative path"
+            result_file_context = next(
+                x for x in gen_data if x.startswith("RESULT_FILE:")
```

Comment:
Would `error_message` be a more meaningful name?

Reply:
Hmm I'm not fully sure about this. This variable is a `ContextValue`, which has context information about the location of the word in the config file. I did type annotate the variable to make it a bit more clear